Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java

```diff
@@ -0,0 +1,41 @@
+package hestia.backend.exceptions;
+
+
+import com.google.gson.JsonObject;
+
+import org.json.JSONObject;
+
+import hestia.backend.exceptions.ServerExceptions.DatabaseException;
+import hestia.backend.exceptions.ServerExceptions.InvalidStateException;
+import hestia.backend.exceptions.ServerExceptions.NotFoundException;
+import hestia.backend.exceptions.ServerExceptions.ServerException;
+import hestia.backend.exceptions.ServerExceptions.SetupFailedException;
+import hestia.backend.exceptions.ServerExceptions.UndefinedServerException;
+
+public class ExceptionFactory {
```

Comment:
Add tests for this class. Note: you should make a test for every case of the switch.